src/Lumper.CLI/Program.cs

```diff
@@ -1,32 +1,182 @@
 
 internal sealed class Program
 {
-    public static void Main(string[] args)
+    private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+
+    private static bool _shouldWriteOutput;
+
+    public static int Main(string[] args)
     {
-        var parser = new Parser(with =>
+        LoggingRule? consoleRule = LogManager.Configuration?.LoggingRules?.FirstOrDefault(rule =>
+            rule.Targets.FirstOrDefault() is ColoredConsoleTarget
+        );
+        if (LogManager.Configuration is null || consoleRule is null)
+        {
+            Console.Error.WriteLine(
+                "Could not find logging configuration! You should have a NLog.config in the same directory as this executable."
+            );
+            return 1;
+        }
+
+        // Remove date and other crap from layout. This is good for file logs but here
+        // we want to just print messages directly.
+        // https://github.com/NLog/NLog/wiki/Layouts
+        consoleRule.Targets.OfType<ColoredConsoleTarget>().First().Layout = "${message:withException=true}";
+
+        ParserResult<CommandLineOptions> parserResult = new Parser(with =>
         {
-            with.HelpWriter = Console.Out;
             with.CaseInsensitiveEnumValues = true;
-        });
-        ParserResult<CommandLineOptions> parserResult = parser.ParseArguments<CommandLineOptions>(args);
+            with.HelpWriter = null;
+            with.AutoHelp = true;
+            with.AutoVersion = true;
+        }).ParseArguments<CommandLineOptions>(args);
+
+        if (parserResult.Errors.Any())
+            return ShowHelp(parserResult);
+
+        CommandLineOptions options = parserResult.Value;
 
-        string? path = null;
-        parserResult.WithParsed(x => path = x.Path).WithNotParsed(CommandLineOptions.ErrorHandler);
-        if (path == null)
+        consoleRule.SetLoggingLevels(options.Verbose ? LogLevel.Debug : LogLevel.Info, LogLevel.Fatal);
+        try
+        {
+            Run(options);
+            return 0;
+        }
+        catch (Exception ex)
+        {
+            Logger.Error(ex);
+            return 1;
+        }
+    }
+
+    // The main application logic of the program, called once the command line options have been parsed.
+    //
+    // It can throw at any point, and the exception will be caught and program will exit with non-zero
+    // status code.
+    //
+    // If method completes successfully, the program will exit with status code 0.
+    private static void Run(CommandLineOptions options)
+    {
+        var bspFile = BspFile.FromPath(options.InputPath, null);
+
+        if (bspFile is null)
+            throw new InvalidDataException("Failed to load BSP file");
+
+        if (options.JobWorkflow is { } workflowPath)
+            RunWorkflow(bspFile, workflowPath);
+
+        if (_shouldWriteOutput || options.Compress || options.DontCompress)
+        {
+            DesiredCompression compression = DesiredCompression.Unchanged;
```

Comment:
This should be `var compression = ...`

Reply:
As above. I'd personally prefer using `var` here, but dotnet code analysis rules don't allow you to make enum values use `var` without making hundreds of other variables use `var` that we don't want to.